checker/bin/spark-checker.sh

```diff
@@ -33,48 +33,94 @@ PARTITIONS
 
 -h  display this help."
 
+ALLUXIO_CHECKER_JAR=""
 PARTITIONS="${2:-10}";
 SPARK_MASTER="$1";
 SPARK_SUBMIT="";
 
 # Find the location of spark-submit in order to run the Spark job
 function find_spark_path() {
-  { # Try SPARK_HOME
-    [ -f "${SPARK_HOME}/bin/spark-submit" ] && SPARK_SUBMIT="${SPARK_HOME}/bin"
-  } || { # Try SPARKPATH
-    [ -f "${SPARKPATH}/spark-submit" ] && SPARK_SUBMIT="${SPARKPATH}"
-  } || { # Try PATH
+  # Try to find spark_submit in ${SPARK_HOME}
+  if [[ "${SPARK_HOME}" != "" ]]; then
+    {
+      [ -f "${SPARK_HOME}/bin/spark-submit" ] && SPARK_SUBMIT="${SPARK_HOME}/bin/spark-submit"
+    } || {
+      if [[ "${SPARK_SUBMIT}" == "" ]]; then
+        array=(`find "${SPARK_HOME}" -type f -name 'spark-submit'`)
+        for i in "${array[@]}"; do
+          SPARK_SUBMIT="$i"
+          break;
+        done
+      fi
+    }
+  fi
+
+  # Try to find spark_submit in ${SPARKPATH}
+  if [[ "${SPARK_SUBMIT}" == "" ]]  &&  [[ "${SPARKPATH}" != "" ]]; then
+    {
+      [ -f "${SPARKPATH}/spark-submit" ] && SPARK_SUBMIT="${SPARKPATH}/spark-submit"
+    } || {
+      if [[ "${SPARK_SUBMIT}" == "" ]]; then
+        array=(`find "${SPARKPATH}" -type f -name 'spark-submit'`)
+        for i in "${array[@]}"; do
+          SPARK_SUBMIT="$i"
+          break;
+        done
+      fi
+    }
+  fi
+
+  # Try to find spark_submit in ${PATH}
+  if [[ "${SPARK_SUBMIT}" == "" ]]  &&  [[ "${PATH}" != "" ]]; then
     IFS=':' read -ra PATHARR <<< "$PATH"
     for p in "${PATHARR[@]}"; do
       if [ -f "$p/spark-submit" ]; then
-        SPARK_SUBMIT="$p"
+        SPARK_SUBMIT="$p/spark-submit"
         break;
       fi
     done
-    if [[ "${SPARK_SUBMIT}" == "" ]]; then
-      echo -e "Please set SPARK_HOME or SPARKPATH before running Spark integration checker." >&2
-      exit 1
-    fi
-  } 
+  fi
+
+  if [[ "${SPARK_SUBMIT}" == "" ]]; then
+    echo -e "Please set SPARK_HOME before running Spark integration checker." >&2
```

Comment:
for this to be correct, we need to enumerate the places that we searched. Otherwise the user might set `$SPARK_HOME` but still get this message

Reply:
If users have set $SPARK_HOME and we do not find any spark-submit file in this  $SPARK_HOME, we echo the message to let users know we cannot find spark-submit in $SPARK_HOME, otherwise, we ask users to set $SPARK_HOME.  You remind me of the possibility that the $SPARK_HOME may be invalid, Thanks!!